src/parser.ts

```diff
@@ -7906,6 +7932,11 @@ export function parseClassDeclaration(
 
   const decorators: ESTree.Decorator[] = context & Context.OptionsNext ? parseDecorators(parser, context) : [];
 
+  if (parser.leadingDecorators) {
+    decorators.unshift(...parser.leadingDecorators);
```

Comment:
Why `unshift`? It's slow

Reply:
concat?

Reply:
Push?

https://dev.to/uilicious/javascript-array-push-is-945x-faster-than-array-concat-1oki